hebsafeharbor/common/date_utils.py

```diff
@@ -60,30 +68,35 @@ def reconstruct_date_string(self) -> str:
         return masked_text
  
 
-def set_date_mention_component(matched:re.Match,ind:int) -> DateMentionComponent:
+def set_date_mention_component(matched:re.Match,ind:Union[int,str]) -> DateMentionComponent:
     '''
     Extracts the date and its span from a regular expression and sets a DateMentionComponent 
     instance accordingly.
     :param matched: a re.Match instance containing the groups of a regular expression
-    :param ind: the index of the group we would like to retrieve
+    :param ind: the index (or group name) of the group we would like to retrieve. if ind<1 (or group name not in the matched groups), return an empty instance
     :return: a DateMentionComponent instance
     '''
-    return DateMentionComponent(matched.group(ind),matched.span(ind)[0],matched.span(ind)[1])
-
-def set_date_mention(matched:re.Match,group_order:List[int]) -> DateMention:
+    if type(ind) == int:
+        date_instance = (
+            DateMentionComponent(matched.group(ind),matched.span(ind)[0],matched.span(ind)[1]) if ind>0 else None
+        )
```

Comment:
please remove parentheses

Reply:
in documentation you said you return empty instance in case that the index / group are invalid but here you return None so either fix it in documentation or change it here whatever you prefer